### PR TITLE
Upgrade to v2.2.5 of twemoji

### LINKED ISSUE #10028: Reader: tiled gallery emoji issues in IE11
Native emoji and twemoji images do not appear as expected in IE11 inside a tiled gallery.

Check out the following post: https://wpcalypso.wordpress.com/read/feeds/25801/posts/1050231304

### Chrome

Native emoji:

screen shot 2016-12-14 at 15 17 23

Twemoji image:
screen shot 2016-12-14 at 15 17 32

### IE11

Native emoji:
screen shot 2016-12-14 at 15 19 43

Twemoji image:
screen shot 2016-12-14 at 15 19 37

### PR DESCRIPTION
**Pull request by blowery**
Update the twemoji library to 2.2.5.

https://github.com/twitter/twemoji

Fixes #10028.

### CODE REVIEW

**Comment by bluefuton** [comment]
@blowery have we updated the corresponding images on the CDN too? I had a peek at wpcom and found @pento's r145912, but this only seems to add the 72x72 images.

**Comment by blowery** [comment]
@bluefuton I'm confused by twemoji. According to the [Breaking Changes in 2.2](https://github.com/twitter/twemoji#breaking-changes-in-v2) there shouldn't _be_ any PNG assets outside of 72x72, but clearly there are icons in there at 36 and 16. It appears they are there for v1? 

Also see https://github.com/twitter/twemoji/blob/gh-pages/2/twemoji.npm.js#L33 which makes me think 72px is the only size supported now?

**Comment by blowery** [comment]
Did a bit more digging on this.

```
blowery@loki ~/src/github/twemoji
$ ls -l 36x36 | wc -l
     873

blowery@loki ~/src/github/twemoji
$ ls -l 72x72 | wc -l
     873

blowery@loki ~/src/github/twemoji
$ ls -l 2/72x72 | wc -l
    2423
```

So the old spots don't have all the new assets.

**Comment by bluefuton** [comment]
Cool - sounds like we only need the 72x72 then 👍

File: config/_shared.json [hunk 1]
```diff
@@ -144,5 +144,5 @@
 		{ "value": 452, "langSlug": "zh-tw", "name": "zh-tw - 繁體中文", "wpLocale": "zh_TW", "popular": 14 }
 	],
 	"project": "wordpress-com",
-	"twemoji_cdn_url": "//s0.wp.com/wp-content/mu-plugins/wpcom-smileys/twemoji/"
```
Comment (samouri) [inline config/_shared.json]: what are we cdning?

Comment (bluefuton) [inline config/_shared.json]: @samouri the emoji images. They live on wpcom and are automagically available on our CDN.

Comment (blowery) [inline config/_shared.json]: the images that get used to represent the emoji

Comment (samouri) [inline config/_shared.json]: so twemoji somehow maps between the ascii and the cdn location? cool

**Comment by samouri** [comment]
All the code config LGTM.  Is there anything in particular worth testing (which case was breaking before)

**Comment by blowery** [comment]
@raoulwegat any chance you could add your repro here? @samouri the problem was mostly on IE11. Other browsers support emoji natively, so they didn't appear to be as broken.

**Comment by bluefuton** [comment]
I think the original issue was https://github.com/Automattic/wp-calypso/issues/10028.

**Comment by raoulwegat** [comment]
@blowery: NP. I'm also adding the new `36x36` folder with new twemoji on the CDN, so the URL will need to change.

**Comment by raoulwegat** [comment]
Static media added to CDN in https://wpcom.trac.automattic.com/changeset/151104

**Comment by raoulwegat** [comment]
Added 36x36 PNGs in r151104-wpcom

**Comment by raoulwegat** [comment]
We need to wait until #11453 is merged before merging this, because IE11 will now show all twemoji, but they'll all appear squashed like this:
screen shot 2017-02-17 at 3 57 58 pm

**Comment by bluefuton** [comment]
#11453 is merged 👍

**Comment by blowery** [comment]
> NP. I'm also adding the new 36x36 folder with new twemoji on the CDN, so the URL will need to change.

I'm ... not sure we want to change it wholesale. `twemoji` allows for the invocation to specify the correct size, which we may want to start taking advantage of. using 36px emoji in reader titles likely won't look very good. We probably want the svg or 72px version there?

**Comment by raoulwegat** [comment]
@blowery: I'm not sure how twemoji decides on which size to use, but it's currently using the 72px size anyway when testing with this branch – even though the 36px are deployed on the CDN.

**Comment by bluefuton** [comment]
@raoulwegat as @blowery mentioned above, I don't think the 36x36 assets are used by Twemoji 2:

> the default PNG icon size is 72 pixels and there are no other PNG assets for 16 or 32.

https://github.com/twitter/twemoji#breaking-changes-in-v2

**Comment by bluefuton** [comment]
In fb150bc, I updated the `<Emojify>` component for twemoji v2 and gave it a spring clean. It's not used in Reader, but several other spots in Calypso make use of it.

I've also added a Devdocs page for the component, so you can try the changes:

http://calypso.localhost:3000/devdocs/design/emojify

cc @samouri

File: client/components/emojify/style.scss [hunk 2]
```diff
@@ -1,10 +1,6 @@
-/**
- * Emojify
- */
-
-// Emoji!!
 .emojify__emoji {
```
Comment (bluefuton) [inline client/components/emojify/style.scss]: These are the styles recommended by the twemoji authors:

https://github.com/twitter/twemoji#inline-styles

The old styles left consecutive emoji looking a bit bunched up.

**Comment by raoulwegat** [comment]
> @raoulwegat any chance you could add your repro here?

@blowery that was already merged in https://github.com/Automattic/wp-calypso/pull/11453

**Comment by samouri** [comment]
@bluefuton: code looks good!

File: client/components/emojify/style.scss (hunk 2 above)
Comment (samouri) [inline client/components/emojify/style.scss]: might be helpful to throw that as a comment in the `scsss` before checking in

Comment (bluefuton) [inline client/components/emojify/style.scss]: Good call - added 👍

**Pull request merged**